floyd/cli/data.py

```diff
@@ -141,6 +143,71 @@ def clone(id):
                               delete_after_untar=True)
 
 
+@click.command()
+@click.argument('dataset_name', nargs=1)
+def listfiles(dataset_name):
+    """
+    List files in the given dataset
+    """
+
+    data_source = DataClient().get(normalize_data_name(dataset_name, use_data_config=False))
+    if dataset_name and not data_source:
+        # Try with the raw ID
+        data_source = DataClient().get(dataset_name)
+
+    if not data_source:
+        if 'output' in dataset_name:
+            floyd_logger.info("Note: You cannot clone the output of a running job. You need to wait for it to finish.")
+        sys.exit()
+
+    # Depth-first search
+    dirs = ['']
+    paths = []
+    while dirs:
+        cur_dir = dirs.pop()
+        url = "/resources/{}/{}?content=true".format(data_source.resource_id, cur_dir)
+        response = DataClient().request("GET", url).json()
+
+        if response['skipped_files'] > 0:
+            floyd_logger.info("Warning: in directory '{}', {}/{} files skipped (too many files)".format(cur_dir, response['skipped_files'], response['total_files']))
```

Comment:
nitpick, to keep the convention consistent, could you change it to use the following logging style?

```python
floyd_logger.info("Warning: in directory '%s', %s/%s files skipped (too many files)", cur_dir, response['skipped_files'], response['total_files'])
```